lib/surface_site_web/views/layout_view.ex

```diff
@@ -6,16 +6,41 @@ defmodule SurfaceSiteWeb.LayoutView do
     <html lang="en">
       <head>
         {{ Phoenix.HTML.Tag.csrf_meta_tag() }}
-        <meta charset="utf-8"/>
-        <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
-        <meta name="viewport" content="initial-scale=1, maximum-scale=1, minimum-scale=1"/>
+        <meta charset="utf-8" />
+        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+        <meta name="viewport" content="initial-scale=1, maximum-scale=1, minimum-scale=1" />
         <title>Surface</title>
-        <link phx-track-static rel="stylesheet" href={{ Routes.static_path(@conn, "/css/app.css") }}/>
-        <script defer phx-track-static type="text/javascript" src={{ Routes.static_path(@conn, "/js/app.js") }}></script>
+        <link phx-track-static rel="stylesheet" href={{ Routes.static_path(@conn, "/css/app.css") }} />
+        <script
+          defer
+          phx-track-static
+          type="text/javascript"
+          src={{ Routes.static_path(@conn, "/js/app.js") }}
+        />
```

Comment:
@paulstatezny I'm pretty sure that some elements like the `<script>` tag require an explicit closing `</script>`. The self-closing syntax is only accepted by XHTML (some browsers ignore that and make it work anyway) but I believe the safest approach is to:

  * Never self-close HTML tags.
  * Never create closing tags for void elements (area, base, br, col, hr, img, input, link, meta, param, command, keygen, source)
  * Create closing tags for all other elements.

There's a big chance the surface compiler already follows those rules so this may not be a problem as the final template will be correctly generated. However, if this changes in the future, it could be a problem.

Reply:
Ah, yes that's a foundation of a few formatter decisions: the fact that Surface (like React) allows you to make self-closing tags and intelligently expands them into opening/closing tags when it renders the actual HTML.

It didn't cross my mind that that might change in the future. (I'm curious why we'd want that?) I don't really have a dog in the fight. I just thought it looked clean.

Also, if it does change in the future, we could always modify the formatter at that point instead of optimizing for a future that we're unsure of when/if it will arrive. Just a thought.